activestorage/lib/active_storage/attached/model.rb

```diff
@@ -141,12 +160,12 @@ def #{name}=(attachables)
                 if Array(attachables).none?
                   ActiveStorage::Attached::Changes::DeleteMany.new("#{name}", self)
                 else
-                  ActiveStorage::Attached::Changes::CreateMany.new("#{name}", self, attachables)
+                  ActiveStorage::Attached::Changes::CreateMany.new("#{name}", self, attachables, key)
```

Comment:
```suggestion
                  ActiveStorage::Attached::Changes::CreateMany.new("#{name}", self, attachables, "#{key}")
```

(This is why the tests are failing)